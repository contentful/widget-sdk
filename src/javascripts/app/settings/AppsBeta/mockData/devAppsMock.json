[
  {
    "sys": {
      "type": "DevApp",
      "id": "dev-app_2NAHYQkIGItuEzwePZlWXV"
    },
    "extensionDefinition": {
      "sys": {
        "id": "2NAHYQkIGItuEzwePZlWXV",
        "type": "ExtensionDefinition",
        "organization": {
          "sys": {
            "type": "Link",
            "linkType": "Organization",
            "id": "2b5UqceLpT58d9tat2os42"
          }
        },
        "createdAt": "2019-07-23T10:00:42.152Z",
        "updatedAt": "2019-07-23T10:00:42.152Z",
        "createdBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "2jvc3kU4n7OIABiFMTaGyB"
          }
        },
        "updatedBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "2jvc3kU4n7OIABiFMTaGyB"
          }
        }
      },
      "name": "Optimizely",
      "src": "http://localhost:1234",
      "public": false,
      "locations": ["page"]
    }
  }
]
