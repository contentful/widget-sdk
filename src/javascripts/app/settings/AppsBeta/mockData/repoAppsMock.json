[
  {
    "sys": {
      "type": "App",
      "id": "netlify"
    },
    "extensionDefinition": {
      "sys": {
        "id": "1VchawWvbIClHuMIyxwR5m",
        "type": "ExtensionDefinition",
        "organization": {
          "sys": {
            "type": "Link",
            "linkType": "Organization",
            "id": "5EJGHo8tYJcjnEhYWDxivp"
          }
        },
        "createdAt": "2019-07-24T09:09:45.574Z",
        "updatedAt": "2019-08-01T14:41:20.716Z",
        "createdBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "46JHc8ru6cDDm0CtthS0Kw"
          }
        },
        "updatedBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "46JHc8ru6cDDm0CtthS0Kw"
          }
        }
      },
      "name": "Netlify build and preview",
      "src": "https://netlify-build-1.contentfulexts.com/extension.html",
      "public": true,
      "locations": ["app", "entry-sidebar"]
    },
    "extension": {
      "sys": {
        "id": "netlify-app-owpoHwFZXAPWpyOy",
        "type": "Extension",
        "space": {
          "sys": {
            "id": "bxgvwg0h5uxs",
            "type": "Link",
            "linkType": "Space"
          }
        },
        "version": 1,
        "createdAt": "2019-08-21T11:21:39.416Z",
        "createdBy": {
          "sys": {
            "id": "2jvc3kU4n7OIABiFMTaGyB",
            "type": "Link",
            "linkType": "User"
          }
        },
        "updatedAt": "2019-08-21T11:21:39.416Z",
        "updatedBy": {
          "sys": {
            "id": "2jvc3kU4n7OIABiFMTaGyB",
            "type": "Link",
            "linkType": "User"
          }
        },
        "environment": {
          "sys": {
            "id": "master",
            "type": "Link",
            "linkType": "Environment"
          }
        }
      },
      "parameters": {
        "names": "thirsty",
        "siteIds": "06fe192d-3ddb-4169-9db7-7c1093017a8c",
        "siteUrls": "https://thirsty-einstein-8908d6.netlify.com",
        "siteNames": "thirsty-einstein-8908d6",
        "buildHookIds": "5d5d2942f6b077dd3cb66e44",
        "notificationHookIds": "5d5d2942d271c00be9a27f6d,5d5d29422ea5fc0f3136bd2a,5d5d2942379672e846227b98"
      },
      "extensionDefinition": {
        "sys": {
          "id": "1VchawWvbIClHuMIyxwR5m",
          "type": "Link",
          "linkType": "ExtensionDefinition"
        }
      }
    }
  },
  {
    "sys": {
      "type": "App",
      "id": "gatsby"
    },
    "extensionDefinition": {
      "sys": {
        "id": "35f8xJFaJpOUFAKepAWiUj",
        "type": "ExtensionDefinition",
        "organization": {
          "sys": {
            "type": "Link",
            "linkType": "Organization",
            "id": "5EJGHo8tYJcjnEhYWDxivp"
          }
        },
        "createdAt": "2019-07-24T12:36:16.933Z",
        "updatedAt": "2019-08-02T12:26:52.709Z",
        "createdBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "46JHc8ru6cDDm0CtthS0Kw"
          }
        },
        "updatedBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "46JHc8ru6cDDm0CtthS0Kw"
          }
        }
      },
      "name": "Gatsby preview",
      "src": "https://gatsby-preview-1.contentfulexts.com/extension.html",
      "public": true,
      "locations": ["app", "entry-sidebar"]
    }
  },
  {
    "sys": {
      "type": "App",
      "id": "optimizely"
    },
    "extensionDefinition": {
      "sys": {
        "id": "i43PggcHag2PCrGjYTiIX",
        "type": "ExtensionDefinition",
        "organization": {
          "sys": {
            "type": "Link",
            "linkType": "Organization",
            "id": "5EJGHo8tYJcjnEhYWDxivp"
          }
        },
        "createdAt": "2019-07-24T12:38:34.885Z",
        "updatedAt": "2019-08-02T13:55:42.195Z",
        "createdBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "46JHc8ru6cDDm0CtthS0Kw"
          }
        },
        "updatedBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "46JHc8ru6cDDm0CtthS0Kw"
          }
        }
      },
      "name": "Optimizely experimentation",
      "src": "https://optimizely-1.contentfulexts.com/extension.html",
      "public": true,
      "locations": ["app", "entry-editor", "entry-sidebar"]
    }
  },
  {
    "sys": {
      "type": "App",
      "id": "cloudinary"
    },
    "extensionDefinition": {
      "sys": {
        "id": "zjcnWgBknf9zB7IM9HZjE",
        "type": "ExtensionDefinition",
        "organization": {
          "sys": {
            "type": "Link",
            "linkType": "Organization",
            "id": "5EJGHo8tYJcjnEhYWDxivp"
          }
        },
        "createdAt": "2019-07-24T12:40:36.439Z",
        "updatedAt": "2019-07-30T17:13:39.053Z",
        "createdBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "46JHc8ru6cDDm0CtthS0Kw"
          }
        },
        "updatedBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "46JHc8ru6cDDm0CtthS0Kw"
          }
        }
      },
      "name": "Cloudinary assets",
      "src": "https://cloudinary-assets-1.contentfulexts.com/extension.html",
      "public": true,
      "locations": ["app", "entry-field"],
      "fieldTypes": [
        {
          "type": "Object"
        }
      ]
    },
    "extension": {
      "sys": {
        "id": "cloudinary-app-eDCiHa2DMfpjoNFH",
        "type": "Extension",
        "space": {
          "sys": {
            "id": "bxgvwg0h5uxs",
            "type": "Link",
            "linkType": "Space"
          }
        },
        "version": 1,
        "createdAt": "2019-08-23T07:00:54.416Z",
        "createdBy": {
          "sys": {
            "id": "29B6zorz8plUCN1auThiVH",
            "type": "Link",
            "linkType": "User"
          }
        },
        "updatedAt": "2019-08-23T07:00:54.416Z",
        "updatedBy": {
          "sys": {
            "id": "29B6zorz8plUCN1auThiVH",
            "type": "Link",
            "linkType": "User"
          }
        },
        "environment": {
          "sys": {
            "id": "master",
            "type": "Link",
            "linkType": "Environment"
          }
        }
      },
      "parameters": {
        "apiKey": "sadfdsaf",
        "maxFiles": 10,
        "cloudName": "sadfsadfdsaf"
      },
      "extensionDefinition": {
        "sys": {
          "id": "zjcnWgBknf9zB7IM9HZjE",
          "type": "Link",
          "linkType": "ExtensionDefinition"
        }
      }
    }
  },
  {
    "sys": {
      "type": "App",
      "id": "bynder"
    },
    "extensionDefinition": {
      "sys": {
        "id": "5KySdUzG7OWuCE2V3fgtIa",
        "type": "ExtensionDefinition",
        "organization": {
          "sys": {
            "type": "Link",
            "linkType": "Organization",
            "id": "5EJGHo8tYJcjnEhYWDxivp"
          }
        },
        "createdAt": "2019-07-24T12:42:27.828Z",
        "updatedAt": "2019-08-01T14:38:17.868Z",
        "createdBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "46JHc8ru6cDDm0CtthS0Kw"
          }
        },
        "updatedBy": {
          "sys": {
            "type": "Link",
            "linkType": "User",
            "id": "46JHc8ru6cDDm0CtthS0Kw"
          }
        }
      },
      "name": "Bynder assets",
      "src": "https://bynder-assets-1.contentfulexts.com/extension.html",
      "public": true,
      "locations": ["app", "entry-field"],
      "fieldTypes": [
        {
          "type": "Object"
        }
      ]
    }
  }
]
