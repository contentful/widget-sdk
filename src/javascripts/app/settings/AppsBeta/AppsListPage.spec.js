import React from 'react';
import { render, wait, cleanup } from '@testing-library/react';
import AppsListPage from './AppsListPage.es6';
import appsListingMock from './mockData/appsListingMock.json';
import devAppsMock from './mockData/devAppsMock.json';
import repoAppsMock from './mockData/repoAppsMock.json';
import * as spaceContextMocked from 'ng/spaceContext';

spaceContextMocked.getData.mockReturnValue(true);

jest.mock('./AppMarkdown.es6', () => () => null, { virtual: true });

jest.mock('MicroBackendsClient.es6', () => {
  return () => {};
});

describe('AppsListPage', () => {
  const orgId = '123';
  const spaceId = '456';
  const userId = '000';
  const mockProductCatalog = {
    loadProductCatalogFlags: jest.fn(() =>
      Promise.resolve({
        basic_apps: true,
        optimizely_app: true
      })
    ),
    isAppsFeatureDisabled: () => Promise.resolve(false),
    isAppEnabled: () => true
  };

  afterEach(cleanup);

  it('should match snapshot for loading state', async () => {
    const mockRepo = {
      getApps: jest.fn(),
      getMarketplaceApps: jest.fn(),
      getDevApps: jest.fn(),
      isDevApp: jest.fn()
    };

    const { container } = render(
      <AppsListPage
        goToContent={() => {}}
        repo={mockRepo}
        organizationId={orgId}
        spaceId={spaceId}
        userId={userId}
        productCatalog={mockProductCatalog}
      />
    );

    expect(container).toMatchSnapshot();
  });
  it('should match snapshot of apps loaded state', async () => {
    const mockRepo = {
      getApps: jest.fn(() => Promise.resolve(repoAppsMock)),
      getMarketplaceApps: jest.fn(() => Promise.resolve(appsListingMock)),
      getDevApps: jest.fn(() => Promise.resolve(devAppsMock)),
      isDevApp: jest.fn()
    };

    const { container } = render(
      <AppsListPage
        goToContent={() => {}}
        repo={mockRepo}
        organizationId={orgId}
        spaceId={spaceId}
        userId={userId}
        productCatalog={mockProductCatalog}
      />
    );

    await wait();

    expect(container).toMatchSnapshot();
  });
});
