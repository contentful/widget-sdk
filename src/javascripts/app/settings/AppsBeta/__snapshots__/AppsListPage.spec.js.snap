// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppsListPage should match snapshot for loading state 1`] = `
.emotion-1 {
  margin-bottom: 1.5rem;
}

.emotion-0 {
  margin-right: 0.75rem;
  background: #3072be;
  color: #ffffff;
  padding: 0.25rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  line-height: 0.65rem;
  font-size: 0.65rem;
  border-radius: 3px;
  text-transform: uppercase;
}

.emotion-4 {
  background-color: #f7f9fa;
}

.emotion-3 {
  position: relative;
}

<div>
  <div
    class="Workbench__Workbench___THi9K emotion-4"
    data-test-id="cf-ui-workbench"
  >
    <div
      class="Workbench__Workbench__header___2wo2E"
      data-test-id="cf-ui-workbench-header"
    >
      <div
        class="Workbench__Workbench__header-icon___1kJlL"
      >
        <span
          class="icon-component"
          data-icon-name="page-apps"
        >
          <svg
            height="34"
            viewBox="0 0 34 34"
            width="34"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M-2 0h36v36H-2z"
              />
              <path
                d="M28.75 16.5H26.5v-6c0-1.65-1.35-3-3-3h-6V5.25c0-2.07-1.68-3.75-3.75-3.75C11.68 1.5 10 3.18 10 5.25V7.5H4a2.996 2.996 0 0 0-2.985 3v5.7H3.25a4.052 4.052 0 0 1 4.05 4.05 4.052 4.052 0 0 1-4.05 4.05H1V30c0 1.65 1.35 3 3 3h5.7v-2.25a4.052 4.052 0 0 1 4.05-4.05 4.052 4.052 0 0 1 4.05 4.05V33h5.7c1.65 0 3-1.35 3-3v-6h2.25c2.07 0 3.75-1.68 3.75-3.75 0-2.07-1.68-3.75-3.75-3.75z"
                fill="#C0F2D6"
                stroke="#19CD91"
                stroke-width="1.5"
              />
            </g>
          </svg>
        </span>
      </div>
      <div
        class="Workbench__Workbench__header-title___3xp7b"
        data-test-id="workbench-title"
      >
        <h1
          class="Heading__Heading___83D3K"
          data-test-id="cf-ui-heading"
        >
          Apps 
          <span
            class="emotion-0"
          >
            Beta
          </span>
        </h1>
      </div>
      <div
        class="Workbench__Workbench__header-actions___mJoSR"
      >
        <a
          class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
          data-test-id="cf-ui-text-link"
          href="https://www.contentful.com/developers/docs/extensibility/apps/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="TabFocusTrap__TabFocusTrap___39Vty"
            tabindex="-1"
          >
            View documentation
          </span>
        </a>
      </div>
    </div>
    <div
      class="Workbench__Workbench__content-wrapper___3yWKi"
    >
      <div
        class="Workbench__Workbench__content___1U8bV"
        data-test-id="cf-ui-workbench-content"
      >
        <div
          class="Workbench__Workbench__content-inner___36_Hx Workbench__Workbench__content-inner--text___7nfTX"
        >
          <div
            class="Note__Note___2eSKN emotion-1 Note__Note--primary___-8-7D"
            data-test-id="cf-ui-note"
          >
            <div
              class="Note__Note__icon___20RqC"
            >
              <svg
                class="Icon__Icon___38Epv Icon__Icon--small___1yGZK undefined"
                data-test-id="cf-ui-icon"
                height="18"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                />
              </svg>
            </div>
            <div>
              <div>
                <button
                  class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
                  data-test-id="cf-ui-text-link"
                  type="button"
                >
                  <span
                    class="TabFocusTrap__TabFocusTrap___39Vty"
                    tabindex="-1"
                  >
                    Share your feedback
                  </span>
                </button>
                 about apps.
              </div>
            </div>
          </div>
          <div
            class="Card__Card___xXJsT emotion-3 Card__Card--padding-large___3Fgal"
            data-test-id="cf-ui-card"
          >
            <p
              class="Paragraph__Paragraph___2aO-9 emotion-1"
              data-test-id="cf-ui-paragraph"
            >
              Contentful apps extend and expand the capabilities of the Contentful web app. You can integrate your favorite third-party services, build better workflows and customize what you can do with Contentful.
            </p>
            <div>
              <svg
                aria-label="Loading apps list..."
                class="SkeletonContainer__SkeletonContainer___23jiu"
                data-test-id="cf-ui-skeleton-form"
                height="200"
                preserveAspectRatio="xMidYMid meet"
                role="img"
                width="600"
              >
                <title>
                  Loading apps list...
                </title>
                <rect
                  clip-path="url(#cf-ui-skeleton-clip-id)"
                  height="100%"
                  style="fill: url(#cf-ui-skeleton-clip-gradient);"
                  width="100%"
                  x="0"
                  y="0"
                />
                <defs>
                  <clippath
                    id="cf-ui-skeleton-clip-id"
                  >
                    <rect
                      height="21"
                      rx="0"
                      ry="0"
                      width="100"
                      x="0"
                      y="0"
                    />
                    <rect
                      height="36"
                      rx="36"
                      ry="36"
                      width="36"
                      y="60"
                    />
                    <rect
                      height="8"
                      rx="0"
                      ry="0"
                      width="240"
                      x="50"
                      y="75"
                    />
                    <rect
                      height="8"
                      rx="0"
                      ry="0"
                      width="90"
                      x="510"
                      y="75"
                    />
                    <rect
                      height="36"
                      rx="36"
                      ry="36"
                      width="36"
                      y="110"
                    />
                    <rect
                      height="8"
                      rx="0"
                      ry="0"
                      width="240"
                      x="50"
                      y="125"
                    />
                    <rect
                      height="8"
                      rx="0"
                      ry="0"
                      width="90"
                      x="510"
                      y="125"
                    />
                    <rect
                      height="36"
                      rx="36"
                      ry="36"
                      width="36"
                      y="160"
                    />
                    <rect
                      height="8"
                      rx="0"
                      ry="0"
                      width="240"
                      x="50"
                      y="175"
                    />
                    <rect
                      height="8"
                      rx="0"
                      ry="0"
                      width="90"
                      x="510"
                      y="175"
                    />
                  </clippath>
                  <lineargradient
                    id="cf-ui-skeleton-clip-gradient"
                  >
                    <stop
                      offset="0%"
                      stop-color="#e5ebed"
                      stop-opacity="1"
                    >
                      <animate
                        attributeName="stop-color"
                        dur="2s"
                        id="animation1"
                        repeatCount="indefinite"
                        values="#e5ebed; #f7f9fa; #e5ebed"
                      />
                    </stop>
                    <stop
                      offset="50%"
                      stop-color="#f7f9fa"
                      stop-opacity="1"
                    >
                      <animate
                        attributeName="stop-color"
                        begin="animation1.begin+0.25s"
                        dur="2s"
                        repeatCount="indefinite"
                        values="#e5ebed; #f7f9fa; #e5ebed"
                      />
                    </stop>
                    <stop
                      offset="100%"
                      stop-color="#e5ebed"
                      stop-opacity="1"
                    >
                      <animate
                        attributeName="stop-color"
                        begin="animation1.begin+0.5s"
                        dur="2s"
                        repeatCount="indefinite"
                        values="#e5ebed; #f7f9fa; #e5ebed"
                      />
                    </stop>
                  </lineargradient>
                </defs>
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AppsListPage should match snapshot of apps loaded state 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 1rem 0;
  border-bottom: 1px solid #e5ebed;
}

.emotion-9:last-child {
  border-bottom: none;
}

.emotion-7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: block;
}

.emotion-4 {
  font-size: 0.75rem;
  color: #b4c3ca;
}

.emotion-6 {
  font-size: 1rem;
  font-weight: 500;
}

.emotion-8 {
  display: block;
}

.emotion-8 button {
  margin-left: 1rem;
}

.emotion-3 {
  border-radius: 5px;
  background-color: #fff;
  box-shadow: 0 2px 3px 0 rgba(0,0,0,0.08);
  padding: 2px;
  margin-right: 0.75rem;
  width: 35px;
  height: 35px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.emotion-32 {
  margin-top: 3px;
  margin-left: 0.5rem;
}

.emotion-1 {
  margin-bottom: 1.5rem;
}

.emotion-0 {
  margin-right: 0.75rem;
  background: #3072be;
  color: #ffffff;
  padding: 0.25rem;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  line-height: 0.65rem;
  font-size: 0.65rem;
  border-radius: 3px;
  text-transform: uppercase;
}

.emotion-39 {
  background-color: #f7f9fa;
}

.emotion-38 {
  position: relative;
}

<div>
  <div
    class="Workbench__Workbench___THi9K emotion-39"
    data-test-id="cf-ui-workbench"
  >
    <div
      class="Workbench__Workbench__header___2wo2E"
      data-test-id="cf-ui-workbench-header"
    >
      <div
        class="Workbench__Workbench__header-icon___1kJlL"
      >
        <span
          class="icon-component"
          data-icon-name="page-apps"
        >
          <svg
            height="34"
            viewBox="0 0 34 34"
            width="34"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M-2 0h36v36H-2z"
              />
              <path
                d="M28.75 16.5H26.5v-6c0-1.65-1.35-3-3-3h-6V5.25c0-2.07-1.68-3.75-3.75-3.75C11.68 1.5 10 3.18 10 5.25V7.5H4a2.996 2.996 0 0 0-2.985 3v5.7H3.25a4.052 4.052 0 0 1 4.05 4.05 4.052 4.052 0 0 1-4.05 4.05H1V30c0 1.65 1.35 3 3 3h5.7v-2.25a4.052 4.052 0 0 1 4.05-4.05 4.052 4.052 0 0 1 4.05 4.05V33h5.7c1.65 0 3-1.35 3-3v-6h2.25c2.07 0 3.75-1.68 3.75-3.75 0-2.07-1.68-3.75-3.75-3.75z"
                fill="#C0F2D6"
                stroke="#19CD91"
                stroke-width="1.5"
              />
            </g>
          </svg>
        </span>
      </div>
      <div
        class="Workbench__Workbench__header-title___3xp7b"
        data-test-id="workbench-title"
      >
        <h1
          class="Heading__Heading___83D3K"
          data-test-id="cf-ui-heading"
        >
          Apps 
          <span
            class="emotion-0"
          >
            Beta
          </span>
        </h1>
      </div>
      <div
        class="Workbench__Workbench__header-actions___mJoSR"
      >
        <a
          class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
          data-test-id="cf-ui-text-link"
          href="https://www.contentful.com/developers/docs/extensibility/apps/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="TabFocusTrap__TabFocusTrap___39Vty"
            tabindex="-1"
          >
            View documentation
          </span>
        </a>
      </div>
    </div>
    <div
      class="Workbench__Workbench__content-wrapper___3yWKi"
    >
      <div
        class="Workbench__Workbench__content___1U8bV"
        data-test-id="cf-ui-workbench-content"
      >
        <div
          class="Workbench__Workbench__content-inner___36_Hx Workbench__Workbench__content-inner--text___7nfTX"
        >
          <div
            class="Note__Note___2eSKN emotion-1 Note__Note--primary___-8-7D"
            data-test-id="cf-ui-note"
          >
            <div
              class="Note__Note__icon___20RqC"
            >
              <svg
                class="Icon__Icon___38Epv Icon__Icon--small___1yGZK undefined"
                data-test-id="cf-ui-icon"
                height="18"
                viewBox="0 0 24 24"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-6h2v6zm0-8h-2V7h2v2z"
                />
              </svg>
            </div>
            <div>
              <div>
                <button
                  class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
                  data-test-id="cf-ui-text-link"
                  type="button"
                >
                  <span
                    class="TabFocusTrap__TabFocusTrap___39Vty"
                    tabindex="-1"
                  >
                    Share your feedback
                  </span>
                </button>
                 about apps.
              </div>
            </div>
          </div>
          <div
            class="Card__Card___xXJsT emotion-38 Card__Card--padding-large___3Fgal"
            data-test-id="cf-ui-card"
          >
            <p
              class="Paragraph__Paragraph___2aO-9 emotion-1"
              data-test-id="cf-ui-paragraph"
            >
              Contentful apps extend and expand the capabilities of the Contentful web app. You can integrate your favorite third-party services, build better workflows and customize what you can do with Contentful.
            </p>
            <div>
              <h2
                class="Heading__Heading___83D3K"
                data-test-id="cf-ui-heading"
              >
                Installed
              </h2>
              <div
                data-test-id="installed-list"
              >
                <div
                  class="emotion-9"
                >
                  <div
                    class="emotion-7"
                    data-test-id="app-title"
                  >
                    <h3
                      class="Heading__Heading___83D3K emotion-6"
                      data-test-id="cf-ui-heading"
                    >
                      <div
                        class="emotion-5"
                        data-test-id="app-details"
                      >
                        <img
                          class="emotion-3"
                          src="//images.ctfassets.net/lpjm8d10rkpy/2dYQ3x6VMNZJgDXYmMzLBx/b338ad4b50cd0ff31e7157671b801b4d/logomark.svg"
                        />
                        <div>
                          Netlify
                          <div
                            class="emotion-4"
                          >
                            Easily build and preview Netlify sites directly from Contentful 
                          </div>
                        </div>
                      </div>
                    </h3>
                  </div>
                  <div
                    class="emotion-8"
                  >
                    <button
                      class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
                      data-test-id="cf-ui-text-link"
                      type="button"
                    >
                      <span
                        class="TabFocusTrap__TabFocusTrap___39Vty"
                        tabindex="-1"
                      >
                        Configure
                      </span>
                    </button>
                    <button
                      class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
                      data-test-id="cf-ui-text-link"
                      type="button"
                    >
                      <span
                        class="TabFocusTrap__TabFocusTrap___39Vty"
                        tabindex="-1"
                      >
                        About
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="emotion-9"
                >
                  <div
                    class="emotion-7"
                    data-test-id="app-title"
                  >
                    <h3
                      class="Heading__Heading___83D3K emotion-6"
                      data-test-id="cf-ui-heading"
                    >
                      <div
                        class="emotion-5"
                        data-test-id="app-details"
                      >
                        <img
                          class="emotion-3"
                          src="//images.ctfassets.net/lpjm8d10rkpy/3ZH89y9VHHd53v4OQ3BG0V/cc62043a88cd5fb80719687f299e7351/cloudinary_icon_for_white_bg.svg"
                        />
                        <div>
                          Cloudinary
                          <div
                            class="emotion-4"
                          >
                            Reference assets from Cloudinary - digital asset management
                          </div>
                        </div>
                      </div>
                    </h3>
                  </div>
                  <div
                    class="emotion-8"
                  >
                    <button
                      class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
                      data-test-id="cf-ui-text-link"
                      type="button"
                    >
                      <span
                        class="TabFocusTrap__TabFocusTrap___39Vty"
                        tabindex="-1"
                      >
                        Configure
                      </span>
                    </button>
                    <button
                      class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
                      data-test-id="cf-ui-text-link"
                      type="button"
                    >
                      <span
                        class="TabFocusTrap__TabFocusTrap___39Vty"
                        tabindex="-1"
                      >
                        About
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <hr />
              <h2
                class="Heading__Heading___83D3K"
                data-test-id="cf-ui-heading"
              >
                Available
              </h2>
              <div>
                <div
                  class="emotion-9"
                >
                  <div
                    class="emotion-7"
                    data-test-id="app-title"
                  >
                    <h3
                      class="Heading__Heading___83D3K emotion-6"
                      data-test-id="cf-ui-heading"
                    >
                      <div
                        class="emotion-5"
                        data-test-id="app-details"
                      >
                        <img
                          class="emotion-3"
                          src="//images.ctfassets.net/lpjm8d10rkpy/3vNJZRjthatucy2mBJMy1u/0e5d501207427780021e7580cc384761/bynder.5e9e457b9d.svg"
                        />
                        <div>
                          Bynder
                          <div
                            class="emotion-4"
                          >
                            Reference assets from Bynder - digital asset management
                          </div>
                        </div>
                      </div>
                    </h3>
                  </div>
                  <div
                    class="emotion-8"
                  >
                    <button
                      class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
                      data-test-id="cf-ui-text-link"
                      type="button"
                    >
                      <span
                        class="TabFocusTrap__TabFocusTrap___39Vty"
                        tabindex="-1"
                      >
                        About
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="emotion-9"
                >
                  <div
                    class="emotion-7"
                    data-test-id="app-title"
                  >
                    <h3
                      class="Heading__Heading___83D3K emotion-6"
                      data-test-id="cf-ui-heading"
                    >
                      <div
                        class="emotion-5"
                        data-test-id="app-details"
                      >
                        <img
                          class="emotion-3"
                          src="//images.ctfassets.net/lpjm8d10rkpy/4X7O4Q0pIgQZNcONoQrQlp/9262ad9a935fa92e9cacd9207ae0a401/optimizely-logo.svg"
                        />
                        <div>
                          Optimizely
                          <div
                            class="emotion-4"
                          >
                            Experimentation made easy
                          </div>
                        </div>
                      </div>
                    </h3>
                  </div>
                  <div
                    class="emotion-8"
                  >
                    <button
                      class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 TextLink__TextLink--primary___2Vc9F"
                      data-test-id="cf-ui-text-link"
                      type="button"
                    >
                      <span
                        class="TabFocusTrap__TabFocusTrap___39Vty"
                        tabindex="-1"
                      >
                        About
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="emotion-9"
                >
                  <div
                    class="emotion-7"
                    data-test-id="app-title"
                  >
                    <h3
                      class="Heading__Heading___83D3K emotion-6"
                      data-test-id="cf-ui-heading"
                    >
                      <div
                        class="emotion-5"
                        data-test-id="app-details"
                      >
                        <span
                          class="icon-component emotion-3"
                          data-icon-name="page-apps"
                        >
                          <svg
                            height="34"
                            viewBox="0 0 34 34"
                            width="34"
                          >
                            <g
                              fill="none"
                              fill-rule="evenodd"
                            >
                              <path
                                d="M-2 0h36v36H-2z"
                              />
                              <path
                                d="M28.75 16.5H26.5v-6c0-1.65-1.35-3-3-3h-6V5.25c0-2.07-1.68-3.75-3.75-3.75C11.68 1.5 10 3.18 10 5.25V7.5H4a2.996 2.996 0 0 0-2.985 3v5.7H3.25a4.052 4.052 0 0 1 4.05 4.05 4.052 4.052 0 0 1-4.05 4.05H1V30c0 1.65 1.35 3 3 3h5.7v-2.25a4.052 4.052 0 0 1 4.05-4.05 4.052 4.052 0 0 1 4.05 4.05V33h5.7c1.65 0 3-1.35 3-3v-6h2.25c2.07 0 3.75-1.68 3.75-3.75 0-2.07-1.68-3.75-3.75-3.75z"
                                fill="#C0F2D6"
                                stroke="#19CD91"
                                stroke-width="1.5"
                              />
                            </g>
                          </svg>
                        </span>
                        <div>
                          Optimizely
                        </div>
                        <div
                          class="Tag__Tag___Y-myd emotion-32 Tag__Tag--primary___2Hk3I"
                          data-test-id="cf-ui-tag"
                        >
                          Private
                        </div>
                      </div>
                    </h3>
                  </div>
                  <div
                    class="emotion-8"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
